### ISSUE TITLE
Nerdctl can't find .env file

### ISSUE THREAD

**Issue by benjaminpreiss** [rancher-sandbox/rancher-desktop#1213]
### Rancher Desktop Version

0.7.1

### Rancher Desktop K8s Version

1.22.5

### What operating system are you using?

Other (specify below)

### Operating System / Build Version

WSL2 debian on Windows 10 professional

### What CPU architecture are you using?

x64

### Actual Behavior

nerdctl compose can not find my .env file even though specifying it via the --env-file option

### Steps to Reproduce

Create a basic docker-compose.yml e.g.

```yml
version: "3.9"
services:
  web:
    build: .
    ports:
      - "5000:5000"
  redis:
    image: "redis:alpine"
```

Create a .env file in the same directory. e.g.

```env
# .env file
PROJECT_NAME=test_project
```

Run command `nerdctl compose -f ./docker-compose.yml config --env-file ./.env`

### Result

I get an error "FATA[0000] Couldn't find env file: ./.env"

### Expected Behavior

The config should be displayed according to the docker-compose.yml

**Comment by ericpromislow** [comment]
Unfortunately, running rancher-desktop 1.0.0 I can't reproduce this issue, running the command from both the WSL and powershell sides. I'm closing this issue now, but if it's still a problem with the current version, and/or there are pertinent messages in the log files, please feel free to reopen it.

**Comment by ericpromislow** [comment]
Great!

**Comment by benjaminpreiss** [comment]
Hey @ericpromislow. On my side, the problem still persists. :(

**Comment by benjaminpreiss** [comment]
I think this issue is somehow related?

https://github.com/rancher-sandbox/rancher-desktop/issues/1279

**Comment by benjaminpreiss** [comment]
I tried to run it with debug:
```bash
nerdctl compose -f ./docker-compose.yml up --build --env-file ./config-dev.env --debug
```

The only thing shown is:

```
DEBU[0000] found CNI isolation plugin
FATA[0000] Couldn't find env file: ./config-dev.env
```

**Comment by ericpromislow** [comment]
Hi, `nerdctl` brings up a problem trying to hit `containerd.sock` when running in `moby` mode.

This doesn't look exactly like #1279 unless there are some environment variables that `nerdctl` picks up that I don't know about.  Are there?

When I run with `--debug` I also see
```
DEBU[0000] found CNI isolation plugin
```

but then it echoes the contents of the combined config file. This is with `nerdctl version 0.16.0` on both macos and Ubuntu/WSL. 

If I run the command from a different directory, I get the error message complaining about the missing `.env` file, even though there's no `docker-compose.yml` either -- it doesn't get to the point of trying to read that file.

Are you running the command in the same directory, or is this directory mounted at a point other than `/mnt/c` or /mnt/wsl` ?

**Comment by jandubois** [comment]
> Hi, `nerdctl` brings up a problem trying to hit `containerd.sock` when running in `moby` mode.

@ericpromislow You cannot use `nerdctl` with `moby`, we do not support this.

@benjaminpreiss already confirmed that they are running with `containerd` though. Can you reproduce the problem in that configuration?

Since this is on Windows, it sounds like it is a problem with `nerdctl-stub`, which will have to translate the Windows path to a Linux path.

**Comment by ericpromislow** [comment]
I was running the command from the Ubuntu side. In Powershell running `.... .\.env` reproduces the issue. Running `... ./.env`   with a forward slash works fine. The `.\docker-compose.yml` part is resolved with either kind of slash.

**Comment by ericpromislow** [comment]
The `--env-file ENVFILE` part needs to appear before `config`. 

It's not obvious from the `nerdctl` help:
``
$ nerdctl compose --help
Compose

Usage:
  nerdctl compose [flags]
  nerdctl compose [command]
Management commands:
Commands:
  build       Build or rebuild services
  config      Validate and view the Compose file
…
Flags:
      --env-file string            Specify an alternate environment file
  -f, --f string                   Alias of --file
…
```

Those two usage lines are not too helpful if you want to specify both flags and a command. Let's see what docker says:
```
$ docker compose --help

Usage:  docker compose [OPTIONS] COMMAND

Docker Compose

Options:
      --ansi string                Control when to print ANSI control characters ("never"|"always"|"auto") (default "auto")
      --env-file string            Specify an alternate environment file.
  -f, --file stringArray           Compose configuration files
...
```

So doing `nerdctl compose -f .\docker-compose.yml --env-file .\.env config` should work.

What's happening is that the windows stub stops processing paths when it hits the `config` directive, as it assumes that there are no more options afterwards. I'm not sure if this is a bug or by design, so leaving this open. But if you put `config` at the end, it should work.

**Comment by benjaminpreiss** [comment]
@ericpromislow You are absolutely right! It works now, when using --env-file in the right place... I was just so confused by the nerdctl docs, maybe I should request clarification on that there!

**Comment by ericpromislow** [comment]
It was also a bug on our side -- our code wasn't handling the path-type arguments correctly when they appeared to the right of the secondary command (in this case `config`). But the `nerdctl` docs should actually nail down the supported CLI syntax.

**Comment by ericpromislow** [comment]
Not merged yet.

**Comment by ericpromislow** [comment]
Closed in #1332
